data/styles/clear/include/Icons.mapcss

```diff
@@ -2492,50 +2492,59 @@ area|z15-[amenity=sanitary_dump_station]
 
 /* 7.1 Parking */
 
-node|z16[amenity=parking],
-area|z16[amenity=parking]
-{icon-image: parking-s.svg;icon-min-distance: 25;}
-
-node|z17-[amenity=parking],
-area|z17-[amenity=parking]
-{icon-image: parking-m.svg;}
-
+node|z16-[amenity=parking],
+area|z16-[amenity=parking],
+{icon-image: parking-s.svg;icon-min-distance: 5;}
 node|z18-[amenity=parking],
-area|z18-[amenity=parking]
-{font-size: 11;}
-
-node|z16-17[amenity=parking][access=permissive],
-node|z16-17[amenity=parking][access=private],
-{icon-image: zero-icon.svg;}
-
-
-node|z17[amenity=parking_entrance],
-area|z17[amenity=parking_entrance]
+area|z18-[amenity=parking],
+{icon-image: parking-m.svg;font-size: 11;icon-min-distance: 10}
+
+/* no icon as these subtypes should display later */
+node|z16-[amenity=parking][access=private],
+area|z16-[amenity=parking][access=private],
+node|z16-[amenity=parking][parking=street_side],
+area|z16-[amenity=parking][parking=street_side],
+node|z16-[amenity=parking][parking=lane],
+area|z16-[amenity=parking][parking=lane],
+/* =no doesn't work in kothic */
+/* node|z16-[amenity=parking][access=no], */
+/* area|z16-[amenity=parking][access=no], */
+{icon-image: ;icon-min-distance: 0;}
+
+node|z16-[amenity=parking][fee?],
+area|z16-[amenity=parking][fee?],
+{icon-image: parking_pay-s.svg;}
+node|z18-[amenity=parking][fee?],
+area|z18-[amenity=parking][fee?],
+{icon-image: parking_pay-m.svg}
+
+node|z17-[amenity=parking][parking=street_side],
+area|z17-[amenity=parking][parking=street_side],
+node|z17-[amenity=parking][parking=lane],
+area|z17-[amenity=parking][parking=lane],
+{icon-image: parking-s.svg;}
+
+node|z17-[amenity=parking_entrance],
+area|z17-[amenity=parking_entrance],
 {icon-image: parking_entrance-m.svg;}
-node|z17[amenity=parking_entrance][access=permissive],
-node|z17[amenity=parking_entrance][access=private]
-{icon-image: zero-icon.svg;}
+node|z17-[amenity=parking_entrance][access=private],
+area|z17-[amenity=parking_entrance][access=private],
+{icon-image: ;icon-min-distance: 0;}
```

Comment:
no need to reset icon-min-distance: as it was not declared for [amenity=parking_entrance] above

and you didn't add `text: none;` to disable the captions :)

Reply:
oops, fixed 👍